content/en/docs/setup/independent/kubelet-integration.md

```diff
@@ -8,113 +8,160 @@ weight: 40
-The ComponentConfig allows the user to specify flags such as the cluster DNS IP with the following configuration format:
 ```yaml
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
 clusterDNS:
 - 10.96.0.10
 ```
 
-## Problem 2: There should be a way to apply instance-specific configuration
+See the
+[API reference for the
+kubelet ComponentConfig](https://godoc.org/k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig#KubeletConfiguration)
+for more information.
+
+### Providing instance-specific configuration details
+
+Some hosts require specific kubelet configurations, due to differences in hardware, operating system,
+networking, or other host-specific parameters. The following list provices a few examples.
+
+- The path to the DNS resolution file, as specified by the `--resolve-conf` kubelet
+  configuration flag, may differ among operating systems, or depending on whethery ou are using
+  `systemd-resolved`. If this path is wrong, DNS resolution will fail on the Node whose kubelet
+  is configured incorrectly.
+
+- The Node API object `.metadata.name` is set to the machine's hostname by default,
+  unless you are using a cloud provider. You can use the `--hostname-override` flag to override the
+  default behavior if you need to specify a Node name different from the machine's hostname.
 
-There are cases where each kubelet has to be configured individually, due to heterogeneous operating systems,
-machine types, and surrounding environments. Here are some examples of instance-specific flags that need to be dynamically
-set depending on the environment:
-- The value of `--resolv-conf` can vary depending on what OS is running on a particular Node. If you are using
-`systemd-resolved` the path should be `/run/systemd/resolve/resolv.conf`, otherwise it can be `/etc/resolv.conf`.
-If this path is wrong, DNS resolution might not work on your Node.
+- Currently, the kubelet cannot automatically detects the cgroup driver used by the CRI runtime,
+  but the value of `--cgroup-driver` must match the cgroup driver used by the CRI runtime to ensure
+  the health of the kubelet.
+ 
+- Depending on the CRI runtime your cluster uses, you may need to specify different flags to the kubelet.
+  For instance, when using Docker, you need to specify flags such as `--network-plugin=cni`, but if you
+  are using an external runtime, you need to specify `--container-runtime=remote` and specify the CRI
+  endpoint using the `--container-runtime-path-endpoint=<path>`.
 
-- By default, the kubelet uses the hostname of the machine as the Node API object `.metadata.name` and unless
-a cloud provider is used. You can specify a value for `--hostname-override` in a case where the Node name should differ
-from the machine hostname.
+You can specify these flags by configuring an individual kubelet's configuration in your process manager,
+such as `systemd`.
```

Comment:
the point here is that normally one has to manage these manually for each kubelet, therefore this was under a "problem" section.

i would modify:
> You can specify these flags by configuring an individual kubelet's configuration in your process manager,
such as `systemd`.

to:
>Normally, you have to specify these flags in the kubelet configuration for individual nodes.